pkg/utils/nodeclaim/nodeclaim.go

```diff
@@ -232,3 +205,46 @@ func UpdateNodeOwnerReferences(nodeClaim *v1beta1.NodeClaim, node *v1.Node) *v1.
 	})
 	return node
 }
+
+// TerminationHandler is a helper function that takes a v1beta1.NodeClaim and calls cloudProvider.Delete() if status condition
+// on nodeClaim is not terminating. If it is terminating then it will check if the kubelet process is still running. It will
+// return an error and a boolean that indicates if the reconciliation should be re-enqueued.
```

Comment:
The boolean should indicate whether the underlaying instance is terminated. This mainly a semantic and wording change for how we see the returned value from this function